src/Om/Repository/AbstractObjectRepository.php

```diff
@@ -78,11 +80,20 @@ public function findByLike(array $criteria, ?array $orderBy = null, ?int $limit
         return $collection;
     }
 
+    private function defineLimit(?int $limit = null)
+    {
+        if ($limit === null) {
+            $limit = $this->count([]);
```

Comment:
do we need to know the total number of results if no limit is indicated?

it adds a call to redis in addition to FT.SEARCH

Reply:
yes, that's what I'm telling you, what we can do is make a constant since FT.SEARCH seems to return 10 results by default, so we can set a constant to 10000 so there are no more count calls, do you prefer that?

Reply:
you're right, I prefer to avoid useless redis call and put an hardcoded value.

Maybe we could add it directly in the attribute #[Entity] ? 10000 by default and configurable for users, what do you think ?

Reply:
from my point of view, I'm not sure it's the Entity attribute that should carry this, I'd say it's a value linked to the ft.search functional, I'd say that either we remove the nullable from the limit argument and set it to 10000 by default, or we define a constant that would be the value to avoid calls to the count method in the AbstractObjectRepository. Knowing that if you have more than 10000 results, which could happen in my case, you'll have to iterate on them automatically when you do a findAll or this kind of method, since you want to have everything. 


What do you think?

Reply:
I agree, but I think users should be able to change the default number.
 I still have no idea of the best way to do it, if you have an idea I'd love to hear it, otherwise let's keep it constant.

Reply:
you can use FT.AGGREGATE and then a FT.CURSOR cursor, but I'm not sure that's a good idea.